packages/compass-import-export/src/modules/import.ts

```diff
@@ -276,45 +287,51 @@ export const startImport = () => {
         delimiter,
         ignoreBlanks,
         stopOnErrors,
+        errorLogFilePath,
         exclude,
         transform,
       }
     );
 
     const abortController = new AbortController();
     const abortSignal = abortController.signal;
-    dispatch(onStarted(abortController));
-
-    let promise;
+    dispatch(
+      onStarted({
+        abortController,
+        errorLogFilePath: errorLogFilePath || '',
+      })
+    );
 
-    // TODO: log file, but probably only useful once we have the toast (COMPASS-6564)
-    //const logPath = path.join(app.getPath('logs'), path.basename(fileName) + '.log');
-    //const output = fs.createWriteStream(logPath);
+    let promise: Promise<ImportResult>;
 
-    const errors: ErrorJSON[] = [];
     const errorCallback = (err: ErrorJSON) => {
-      errors.push(err);
+      if (errors.length < 5) {
+        // Only store the first few errors in memory.
+        // The log file tracks all of them.
+        // If we are importing a massive file with many errors we don't
+        // want to run out of memory.
+        errors.push(err);
+      }
     };
 
+    let importDone = false;
+
+    // Note: This progress callback can be called
+    // after the import has completed/aborted.
```

Comment:
Not sure I follow why would we want that 🤔 throttle returns both flush and cancel methods on the returned throttled function, seems like in case of abort we would want to cancel any progress updates, or am I missing something?

Reply:
I might have written this before I flushed the throttled function 😅 . I'll check again and make sure it's happening as expected and remove this extra flag then.

Reply:
Ah, gotcha, missed the flush at the end, left a comment there